pages/api/download.ts

```diff
@@ -0,0 +1,20 @@
+import { NextApiHandler } from 'next';
+import fetch from 'node-fetch';
+
+const handler: NextApiHandler = async (req, res) => {
+  const url = req.query.url;
+  const fileName = decodeURIComponent(req.query.title as string).replace(
```

Comment:
i think you should add some defensive logic for `req.query.url` and `req.query.title` and bail with a 400 status code if you don't have both, rather than do the typescript casts here.